pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -272,3 +280,38 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 }
 
 func (alloc *Action) UnInitialize() {}
+
+func setPodVolumesByNode(pod *v1.Pod, node *v1.Node) *volumescheduling.PodVolumes {
+	client := fake.NewSimpleClientset()
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	opts := []runtime.Option{
+		runtime.WithClientSet(client),
+		runtime.WithInformerFactory(informerFactory),
+	}
+	fh, err := runtime.NewFramework(nil, nil, nil, opts...)
+	if err != nil {
+		klog.V(3).Infof("new framework handle instance failed for %v", err.Error())
+		return nil
+	}
+	pl, err := volumebinding.New(&config.VolumeBindingArgs{
+		BindTimeoutSeconds: 300,
+	}, fh)
+	if err != nil {
+		klog.V(3).Infof("new volumebinding instance failed for %v", err.Error())
+		return nil
+	}
+
+	p := pl.(*volumebinding.VolumeBinding)
+	var podVolumes *volumescheduling.PodVolumes
+	boundClaims, claimsToBind, _, err := p.Binder.GetPodVolumes(pod)
+	if err != nil {
+		klog.V(3).Infof("Get podVolumes for pod %v on node %v failed for %v", pod.Name, node.Name, err.Error())
+		return podVolumes
+	}
+	podVolumes, _, err = p.Binder.FindPodVolumes(pod, boundClaims, claimsToBind, node)
+	if err != nil {
+		klog.V(3).Infof("Find podVolumes for pod %v on node %v failed for %v", pod.Name, node.Name, err.Error())
+		return podVolumes
```

Comment:
no need return podVolumes in `if` scope, because it just print a log and beyond `if` it return podVolumes finally.

Reply:
That's right. Let me optimize it.